### PR TITLE
Tweak handling of subnets

### PR DESCRIPTION
**Pull request by tomverran**
Hi, we're using this module to deploy https://github.com/ovotech/carbon-zero-api/pull/5 and since that is going to live behind a Kong proxy we need the following changes:

- Make it possible to have an internal ALB
- Rename public/private subnets to make their roles clearer given the above
- Remove Graylog egress rule
- Loads of terraform fmt whitespace changes, sorry 😓 

These are breaking changes but hopefully renaming the variables isn't too hard when migrating betwen versions, I kind of think it is better to have clearer variable names than it is to maintain compatibility

### CODE REVIEW

File: nsg.tf [hunk 1]
```diff
@@ -50,14 +50,3 @@ resource "aws_security_group_rule" "nsg_task_egress_rule" {
   security_group_id = aws_security_group.nsg_task.id
 }
 
-resource "aws_security_group_rule" "egress_to_graylog_rule" {
```
Comment (nabil-boag) [inline nsg.tf]: Can you please remove all the graylog stuff from this PR? @kelemensanyi is currently testing this with a live app.

File: example/main.tf [hunk 2]
```diff
@@ -9,35 +9,34 @@ module "fargate_app" {
...
   tags = {
-    application       = var.app_name
-    environment       = var.environment
-    team              = var.team
+    application = var.app_name
+    environment = var.environment
+    team        = var.team
   }
-  docker_image        = var.docker_image
-  graylog_cidr        = []
```
Comment (nabil-boag) [inline example/main.tf]: Remove this, will be in another commit.

**Comment by tomverran** [comment]
@nabil-boag Graylog removed ✨

File: lb.tf [hunk 3]
```diff
@@ -31,12 +31,17 @@ variable "lb_access_logs_expiration_days" {
   default = "3"
 }
 
+variable "lb_internal" {
+  default = false
```
Comment (nabil-boag) [inline lb.tf]: Do we want to add this variable to the docs?

File: lb.tf [hunk 4]
```diff
@@ -31,12 +31,17 @@ variable "lb_access_logs_expiration_days" {
...
+}
+
 resource "aws_alb" "main" {
   name = "${var.app}-${var.environment}"
 
   # launch lbs in the public subnet
-  subnets = split(",", var.public_subnets)
+  subnets         = split(",", var.load_balancer_subnets)
   security_groups = [aws_security_group.nsg_lb.id]
+  internal        = var.lb_internal
```
Comment (mwz) [inline lb.tf]: This prevents the LB from getting a public IP assigned, which is what we need in our setup - would be good to expose that setting in the configuration.

File: lb.tf (hunk 3 above)
Comment (mwz) [reply lb.tf]: I'll do this in the absence of Tom.

**Comment by mwz** [comment]
This can be closed in favour of #20.
